apps/codeforafrica/src/lib/index.js

```diff
@@ -655,7 +661,72 @@ function getProjectTags(options = { includeAll: true }) {
   return tags;
 }
 
-function getHomePageStaticProps() {
+function paginateResults(items, page, pageSize) {
+  // We need to initialize to null for serialization.
+  let count = null;
+  let results = [];
+  let pageNumber = null;
+  let pageSizeNumber = null;
+  if (items?.length) {
+    // Need to ensure page, pageSize are numbers and not strings
+    pageNumber = Number.parseInt(page, 10) || 1;
+    pageSizeNumber = Number.parseInt(pageSize, 10) || 6;
+    count = Math.ceil(items.length / pageSizeNumber);
+    results = items.slice(
+      (pageNumber - 1) * pageSizeNumber,
+      pageNumber * pageSizeNumber
+    );
+  }
+  return {
+    pagination: {
+      count,
+      page: pageNumber,
+      pageSize: pageSizeNumber,
+    },
+    results,
+  };
+}
+
+export async function getStories(options) {
+  const {
+    tag: originalTag,
+    page,
+    "page-size": pageSize = 10,
+    q,
+  } = options || {};
+
+  const tag = originalTag || ALL_TAG;
+
+  let found;
+
+  // if tag is not provided or set to ALL, we will return all stories
+  if (tag === ALL_TAG) {
+    // return all stories
+    found = await getPostsByPrimaryTag("stories");
+  } else {
+    found = await getPostsByTag(tag, options);
+  }
+
+  // append /stories/ to the slug && transform published date to ISO format
+  found = found.map((a) => {
+    const { slug: originalSlug, ...other } = a;
+    const slug = `/stories/${originalSlug}`;
+    return { slug, ...other };
```

Comment:
1. `slug` should remain `slug`, create a new key `href`.
2. this can be done as part of `transformPost` by using `primaryTag` value.